api/src/main/java/org/apache/iceberg/TableScan.java

```diff
@@ -147,8 +147,25 @@ default TableScan select(String... columns) {
    * @param toSnapshotId read append data up to this snapshot id
    * @return a table scan which can read append data from {@code fromSnapshotId}
    * exclusive and up to {@code toSnapshotId} inclusive
+   *
+   * @deprecated Please use the appendsBetween with non-primitive fromSnapshotId of Long type
+   */
+  @Deprecated
+  default TableScan appendsBetween(long fromSnapshotId, long toSnapshotId) {
+    return appendsInRange(Long.valueOf(fromSnapshotId), toSnapshotId);
+  }
+
+  /**
+   * Create a new {@link TableScan} to read appended data from {@code fromSnapshotId} exclusive to {@code toSnapshotId}
+   * inclusive.
+   *
+   * @param fromSnapshotId the last snapshot id read by the user, exclusive. If it is null,
+   *                       all the ancestor snapshots of the toSnapshotId should be discovered
+   * @param toSnapshotId read append data up to this snapshot id
+   * @return a table scan which can read append data from {@code fromSnapshotId}
+   * exclusive and up to {@code toSnapshotId} inclusive
    */
-  TableScan appendsBetween(long fromSnapshotId, long toSnapshotId);
+  TableScan appendsInRange(Long fromSnapshotId, long toSnapshotId);
```

Comment:
We can't use the same method name. otherwise, we ran into the following resolution error for method overloading.

```
/home/runner/work/iceberg/iceberg/spark/v2.4/spark/src/main/java/org/apache/iceberg/spark/source/Reader.java:172: error: reference to appendsBetween is ambiguous

        scan = scan.appendsBetween(startSnapshotId, endSnapshotId);
                   ^
  both method appendsBetween(long,long) in org.apache.iceberg.TableScan and method appendsBetween(java.lang.Long,long) in org.apache.iceberg.TableScan match
```

Reply:
Another naming option is `appendsBetweenScan`